packages/core/src/app/shipping/isSelectedShippingOptionValid.ts

```diff
@@ -0,0 +1,17 @@
+import { Consignment } from '@bigcommerce/checkout-sdk';
+import { every } from 'lodash';
+
+export default function isSelectedShippingOptionValid(consignments: Consignment[]): boolean {
+    if (!consignments.length) {
+        return false;
+    }
+
+    return every(
+        consignments,
+        (consignment) =>
+        (consignment.availableShippingOptions &&
+            consignment.availableShippingOptions.filter(
+                ({ id }) => id === consignment.selectedShippingOption?.id,
+            ).length)
```

Comment:
PR looks good.

Also, I can see this won’t be an issue for the custom shipping method where `availableShippingOptions` is always undefined, but the shipping step is not editable.